python/device/iothub-client-sample/iothub_client_sample.py

```diff
@@ -0,0 +1,166 @@
+        if (result == MAP_OK) & (propertyCount.value > 0):
+            print ("propertyCount: %d" % propertyCount.value)
+            keyArray = c_char_p_array(propertyCount.value).from_address(keys.value)
+            valArray = c_char_p_array(propertyCount.value).from_address(values.value)
+            for i in xrange(0,propertyCount.value):
+                print ("(%s,%s)" % (keyArray[i].value, valArray[i].value))
+    counter.contents.value += 1
+    return IOTHUBMESSAGE_ACCEPTED
+
+def SendConfirmationCallback( result, messageId):
+    print ("Confirmation received for message tracking id = %d with result = %d" % (messageId, result))
+    return IoTHubMessage_Destroy(messageHandle[messageId]);
+
+def iothub_client_sample_http_run():
+
+    # callbacks for c dll
+    message_callback = cast(IOTHUBCALLBACK(ReceiveMessageCallback), c_void_p)
+    send_callback = cast(SENDCALLBACK(SendConfirmationCallback), c_void_p)
+
+    # iothub prep code 
+    iotHubClientHandle = IoTHubClient_LL_CreateFromConnectionString(c_char_p(connectionString), Protocol)
+    if iotHubClientHandle==NULL:
+        raise IotHubError("ERROR: iotHubClientHandle is NULL!")
+
+    if Protocol == HTTP_Protocol:
+        result = IoTHubClient_LL_SetOption( iotHubClientHandle, c_char_p("timeout"), byref(timeout))
+        if result != IOTHUB_CLIENT_OK:
+            print "failure to set option \"timeout\""
+
+        result = IoTHubClient_LL_SetOption( iotHubClientHandle, c_char_p("MinimumPollingTime"), byref(minimumPollingTime))
+        if result != IOTHUB_CLIENT_OK:
+            print "failure to set option \"MinimumPollingTime\""
+
+    result = IoTHubClient_LL_SetMessageCallback( iotHubClientHandle, message_callback, byref(receiveContext));
+    if result != IOTHUB_CLIENT_OK:
+        raise IotHubError("ERROR: IoTHubClient_LL_SetMessageCallback..........FAILED!")
+
+    # iothub send a few messages
+    for i in xrange(0, message_count):
+        msgTxtFormatted = msgTxt % (avgWindSpeed + (random.random() * 4 + 2))
+        messageHandle[i] = IoTHubMessage_CreateFromString(c_char_p(msgTxtFormatted))
+        if messageHandle[i] == NULL:
+            raise IotHubError("ERROR: iotHubMessageHandle is NULL!");
+        propMap = IoTHubMessage_Properties(messageHandle[i])
+        propText = "PropMsg_%d" % i
+        result = Map_AddOrUpdate( propMap, c_char_p("PropName"), c_char_p(propText))
+        if result != MAP_OK:
+            raise IotHubError("ERROR: Map_AddOrUpdate Failed!");
+        result = IoTHubClient_LL_SendEventAsync(iotHubClientHandle, messageHandle[i], send_callback, i)
+        if result != IOTHUB_CLIENT_OK:
+            raise IotHubError("ERROR: IoTHubClient_LL_SendEventAsync..........FAILED!")
+        else:
+            print ("IoTHubClient_LL_SendEventAsync accepted message [%d] for transmission to IoT Hub." % i);
+
+    # Wait for Commands 
+    m = 0
+    while m < 10000: # *0.1s = 1000s
+        IoTHubClient_LL_DoWork(iotHubClientHandle)
```

Comment:
convenience layer is yet causing some problems in callbacks to python, so far it only works with LL layer